inc/class-component.php

```diff
@@ -0,0 +1,618 @@
+
+		// Convert to array if necessary.
+		if ( is_string( $themes ) ) {
+			$themes = [ $themes ];
+		}
+
+		// Remove as needed.
+		foreach ( $themes as $theme ) {
+			if ( $this->theme_options[ $theme ] ) {
+				unset( $this->theme_options[ $theme ] );
+			}
+		}
+
+		// Sanitize the entire thing.
+		$this->sanitize_theme_options();
+
+		return $this;
+	}
+
+	/**
+	 * Loop through the theme options, ensuring they're sanitizied.
+	 *
+	 * @return self
+	 */
+	public function sanitize_theme_options(): self {
+
+		$this->theme_options = array_unique(
+			array_filter(
+				array_map(
+					function( $theme ) {
+						$theme = (string) $theme;
+						$theme = trim( $theme );
+						// @todo Should this be camelcased?
+						return $theme;
+					},
+					$this->theme_options
+				)
+			)
+		);
+
+		return $this;
+	}
+
+	/**
+	 * Get the context provider.
+	 *
+	 * @return array
+	 */
+	public function get_context_provider(): array {
+		return $this->context_provider;
+	}
+
+	/**
+	 * Set the context provider.
+	 *
+	 * @param array $context_provider Context provider.
+	 * @return self
+	 */
+	public function set_context_provider( array $context_provider ): self {
+		$this->context_provider = $context_provider;
+		return $this;
+	}
+
+	/**
+	 * Get the context consumer.
+	 *
+	 * @return array
+	 */
+	public function get_context_consumer(): array {
+		return $this->context_consumer;
+	}
+
+	/**
+	 * Set the context consumer.
+	 *
+	 * @param array $context_consumer Context consumer.
+	 * @return self
+	 */
+	public function set_context_consumer( array $context_consumer ): self {
+		$this->context_consumer = $context_consumer;
+		return $this;
+	}
+
+	/**
+	 * Run a user callback on this class. This can be used to create a fork in
+	 * the method chain.
+	 *
+	 * @param callable $callable Callable.
+	 * @param mixed    ...$args  Optional args to pass to the callback.
+	 * @return function
+	 */
+	public function callback( $callable, ...$args ) {
+		return call_user_func_array( $callable, array_merge( [ &$this ], $args ) );
+	}
+
+	/**
+	 * Execute the callback method in this component's registration.
+	 */
+	public function execute_registry_callback(): self {
+
+		// Check the component registry.
+		$registered_component = \WP_Irving\get_registry()->get_registered_component( $this->get_name() );
+
+		if ( is_null( $registered_component ) || ! is_callable( $registered_component['callback'] ?? '' ) ) {
+			return $this;
+		}
+
+		call_user_func_array( $registered_component['callback'], [ $this ] );
+
+		return $this;
+	}
+
+	/**
+	 * Convert all array keys to camel case.
+	 *
+	 * @param array $array Array to convert.
+	 * @return array Updated array with camel-cased keys.
+	 */
+	public function camel_case_keys( $array ): array {
+
+		// Setup for recursion.
+		$camel_case_array = [];
+
+		// Loop through each key.
+		foreach ( $array as $key => $value ) {
+			// Only return keys that are white-listed. Leave $whitelist empty
```

Comment:
Let's take this opportunity to remove non-inclusive "white-list/black-list" terminology in favor of "allowed/blocked" terms. It doesn't seem like there even is a `whitelist` property for this class at the moment, so this will break.